File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int sum = 0;
        int pro = 1;
        int m = n;
        while (n) {
            sum += n % 10;
            pro *= n % 10;
            n /= 10;
        }
        sum += pro;
        return (m % sum == 0);
    }
};

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.js
/**
 * @param {number} n
 * @return {boolean}
 */
var checkDivisibility = function (n) {
    let sum = 0;
    let pro = 1;
    let m = n;
    while (n) {
        sum += n % 10;
        pro *= n % 10;
        n = Math.floor(n / 10);
    }
    sum += pro;
    return (m % sum === 0);
};